webdav/lock.go

```diff
@@ -68,6 +69,10 @@ type LockSystem interface {
 	// contain whitespace.
 	Create(now time.Time, details LockDetails) (token string, err error)
 
+	// Lockdetails is used to discover lockdetails on a file if it exists
+	// if not implemented, return nil lockDetails
+	LockDetails(path string, fi os.FileInfo) (ld *LockDetails, err error)
```

Comment:
If you want to keep this compatible with external implementations, you may want to consider doing what I did with LockDeleter. You can query whether an object implements the interface, and choose to not call it if it does not.

I.e. where we call LockDetails now, you can do:

```
if detailGetter, ok := ls.(LockDetailGetter); ok {
  lockDetails, err := detailGetter.LockDetails(....)
}
```

While not necessary, it does mean if somebody chooses to take this change for whatever reason they'll find it easier to integrate.